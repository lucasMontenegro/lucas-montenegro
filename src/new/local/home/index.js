import React from "react"
import Typography from "@material-ui/core/Typography"
import createBaseClient from "new/local/paperbase/createBaseClient"
import useStyles from "new/local/home/useStyles"
import translations from "new/local/home/translations"
export default function home (appName, routing) {
  const BaseClient = createBaseClient({
    appName,
    clientName: `home`,
    routing,
  })
  function Home (props) {
    const classes = useStyles()
    return (
      <BaseClient
        {...props}
        subtitles={translations.subtitles}
        icons={translations.icons}
      >
        <div className={classes.root}>
          <Typography variant="h5" component="h1">
            {translations.messages[props.languageCode]}
          </Typography>
        </div>
      </BaseClient>
    )
  }
  Home.propTypes = createBaseClient.propTypes
  return Home
}