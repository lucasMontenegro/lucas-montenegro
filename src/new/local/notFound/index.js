import React from "react"
import createBaseClient from "new/local/paperbase/createBaseClient"
import NotFoundView from "new/local/notFound/NotFoundView"
import useReferrer from "new/local/notFound/useReferrer"
export default function notFound (appName, routing) {
  const BaseClient = createBaseClient({
    appName,
    clientName: `notFound`,
    routing,
  })
  function NotFound (props) {
    const { languageCode, match, location } = props
    const referrer = useReferrer(match, location)
    return (
      <NotFoundView
        languageCode={languageCode}
        match={match}
        referrer={referrer}
        routing={routing}
        BaseClient={BaseClient}
        clientProps={props}
      />
    )
  }
  NotFound.propTypes = createBaseClient.clientPropTypes
  return NotFound
}