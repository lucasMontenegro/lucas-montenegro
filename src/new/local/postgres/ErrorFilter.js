import monitor from "pg-monitor"
export default function ErrorFilter () {
  const matchers = []
  return {
    ignore (fn) {
      matchers.push(fn)
    },
    logger (err, e) {
      // usage:
      // pg.ignoreDbError(function matchUniqueUsernameError (err, e) {
      //   return (
      //     typeof err === `object`
      //     && err.code == `23505`
      //     && err.table == `usernames`
      //     && err.constraint == `usernames_pkey`
      //   )
      // })
      // pg.ignoreDbError === logger
      if (!matchers.find(fn => fn(err, e))) {
        monitor.error(err, e)
      }
    },
  }
}