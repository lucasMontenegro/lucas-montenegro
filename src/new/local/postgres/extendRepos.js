// Extend "db" object, i.e.:
// import Repo from "new/local/Repo"
export default function extendRepos (db, dc) {
  // cont.:
  // db.repo = new Repo(db, dc)
}