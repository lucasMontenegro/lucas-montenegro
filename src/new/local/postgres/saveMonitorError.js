const os = require("os")
const fs = require("fs")
const path = require("path")
const { default: simpleErrorCallback } = require("new/local/utils/simpleErrorCallback")
const { default: globals } = require("new/local/utils/globals")
const logFile = path.resolve(globals.process.env.NODE_PATH, `../pg-errors.log`)
exports.default = function saveMonitorError (msg, info) {
  let logText = os.EOL + msg
  if (info.time) {
    logText = os.EOL + logText
  }
  fs.appendFile(logFile, logText, simpleErrorCallback)
}