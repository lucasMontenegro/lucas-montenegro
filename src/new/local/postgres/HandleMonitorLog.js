import saveMonitorError from "new/local/postgres/saveMonitorError"
import isProduction from "new/local/utils/isProduction"
export default function HandleMonitorLog () {
  const production = isProduction()
  return function handleMonitorLog (msg, info) {
    if (info.event === `error`) {
      production || saveMonitorError(msg, info)
    } else if (production) {
      info.display = false
    }
  }
}