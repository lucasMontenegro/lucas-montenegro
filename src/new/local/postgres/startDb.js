import pgPromise from "pg-promise"
import monitor from "pg-monitor"
import Bluebird from "bluebird"
import HandleMonitorLog from "new/local/postgres/HandleMonitorLog"
import ErrorFilter from "new/local/postgres/ErrorFilter"
import extendRepos from "new/local/postgres/extendRepos"
import globals from "new/local/utils/globals"
import isProduction from "new/local/utils/isProduction"
export default function startDb () {
  const errorFilter = ErrorFilter()
  const initOptions = {
    promiseLib: Bluebird,
    error: errorFilter.logger,
    extend: extendRepos,
  }
  const pgp = pgPromise(initOptions)
  const db = pgp({
    host: globals.process.env.PG_HOST,
    database: globals.process.env.PG_DATABASE,
    user: globals.process.env.PG_USER,
    password: globals.process.env.PG_PASSWORD,
  })
  monitor.setTheme(`matrix`)
  monitor.setLog(HandleMonitorLog())
  monitor.attach(initOptions, (
    isProduction() ? [] :
    [`connect`, `disconnect`, `task`, `transact`, `query`]
  ))
  return { db, pgp, ignoreError: errorFilter.ignore }
}