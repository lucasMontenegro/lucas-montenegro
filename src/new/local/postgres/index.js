import startDb from "new/local/postgres/startDb"
export const { db, pgp, ignoreError } = startDb()
export function closeDb () {
  db.$config.pgp.end()
}