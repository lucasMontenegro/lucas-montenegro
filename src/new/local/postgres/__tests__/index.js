import startDb from "new/local/postgres/startDb"
import * as postgres from "new/local/postgres"
jest.mock(`new/local/postgres/startDb`, () => {
  const values = {
    db: {
      $config: { pgp: { end: jest.fn() } }
    },
    pgp: {},
    ignoreError: {},
  }
  return { __esModule: true, default: jest.fn(() => values) }
})
const startDbValues = startDb()
describe(`new/local/postgres`, () => {
  it(`should re-export`, () => {
    expect(startDb.mock.calls).toMatchSnapshot()
    expect(postgres).toHaveProperty(`db`, startDbValues.db)
    expect(postgres).toHaveProperty(`pgp`, startDbValues.pgp)
    expect(postgres).toHaveProperty(`ignoreError`, startDbValues.ignoreError)
  })
  it(`should close the database`, () => {
    postgres.closeDb()
    expect(startDbValues.db.$config.pgp.end.mock.calls).toEqual([[]])
  })
})