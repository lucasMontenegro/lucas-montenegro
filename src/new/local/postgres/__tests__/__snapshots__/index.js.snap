// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/postgres should re-export 1`] = `
Array [
  Array [],
  Array [],
]
`;
